curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md

```diff
@@ -17,439 +17,507 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --text--
 
-Placeholder question
+What is recursion in programming?
 
 #### --distractors--
 
-Placeholder distractor 1
+A programming technique where a function calls itself.
 
 ---
 
-Placeholder distractor 2
+A method for sorting arrays.
 
 ---
 
-Placeholder distractor 3
+A way to optimize loops.
 
 #### --answer--
 
-Placeholder answer
+A programming technique where a function calls itself.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will be the output of the following function call: `fibonacci(5)`?
+
+```cpp
+int fibonacci(int n) {
+    if (n <= 1) return n;
+    return fibonacci(n - 1) + fibonacci(n - 2);
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+0
 
 ---
 
-Placeholder distractor 2
+3
 
 ---
 
-Placeholder distractor 3
+5
 
 #### --answer--
 
-Placeholder answer
+5
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In the following code, what is the base case?
+
+```cpp
+int factorial(int n) {
+    if (n == 0) return 1;
+    return n * factorial(n - 1);
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+`if (n == 0)`
 
 ---
 
-Placeholder distractor 2
+`return n * factorial(n - 1);`
 
 ---
 
-Placeholder distractor 3
+`return 1;`
 
 #### --answer--
 
-Placeholder answer
+`if (n == 0)`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In the following code, what does the recursive call represent?
+
+
+```cpp
+void printNumbers(int n) {
+    if (n > 0) {
+        printNumbers(n - 1);
+        cout << n << " ";
+    }
+}
+```
```

Comment:
````suggestion
In the following code, what does the recursive call represent?

```cpp
void printNumbers(int n) {
    if (n > 0) {
        printNumbers(n - 1);
        cout << n << " ";
    }
}
```
````
please also use JavaScript

Reply:
Respected maintainers ,All the changes have been made as requested
Please review it and tell me if there's any issue further .

Reply:
This comment has not been resolved.